### PR TITLE
Documented the missing YAML flags

### PR DESCRIPTION
**Pull request by javiereguiluz**
This finishes #6226.

### CODE REVIEW

File: components/yaml.rst [hunk 1]
```diff
@@ -267,6 +267,29 @@ representation of the object.
     parsers will likely not recognize the ``php/object`` tag and non-PHP
     implementations certainly won't - use with discretion!
 
+Parsing and Dumping Objects as Maps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.2
+    Support for parsing and dumping objectas as maps was introduced in Symfony 3.2.
```
Comment (xabbuh) [inline components/yaml.rst]: objects

File: components/yaml.rst [hunk 2]
```diff
@@ -267,6 +267,29 @@ representation of the object.
...
+
+You can dump objects as Yaml maps by using the ``DUMP_OBJECT_AS_MAP`` flag::
+
+    $object = new \stdClass();
+    $object->foo = 'bar';
+
+    $dumped = Yaml::dump(array('data' => $object), 2, 4, Yaml::DUMP_OBJECT_AS_MAP);
+    // $dumped = "data:\nfoo:\nSbar"
```
Comment (xabbuh) [inline components/yaml.rst]: `data:\n    foo:\nbar` because of the indentation, right?

File: components/yaml.rst [hunk 3]
```diff
@@ -267,6 +267,29 @@ representation of the object.
...
+
+And parse them by using the ``PARSE_OBJECT_FOR_MAP`` flag::
+
+    $parsed = Yaml::parse($dumped, Yaml::PARSE_OBJECT);
+    var_dump(is_object($parsed)); // true
+    echo $parsed->foo; // bar
```
Comment (xabbuh) [inline components/yaml.rst]: if `$dumped` has the same value as above, we would need to account for `data` here or omit it above while dumping

**Comment by javiereguiluz** [comment]
@xabbuh thanks for the review. To be 100% sure about the code, I created an snippet and run it to double check the results.

**Pull request closed without merging**